src/Chapter2.hs

```diff
@@ -348,8 +354,12 @@ Implement a function that returns only the first half of a given list.
 >>> firstHalf "bca"
 "b"
 -}
--- PUT THE FUNCTION TYPE IN HERE
-firstHalf l = error "firstHalf: Not implemented!"
+firstHalf :: [a] -> [a]
+firstHalf [] = []
+firstHalf (x:xs)  | l > 0 = x : firstHalf xs
+                  | otherwise = []
+                  where l = div (length (x:xs) - 1) 2
```

Comment:
`length (x:xs) - 1` is the same as `length xs` 🙂 

But generally note, that this solution would be slow, as `length` function on lists in Haskell is not cheap. It walks each element of the list in order to get the length. So when you recursively ask `length` on each interaction, this would be too expensive 😞 

Luckily, here, you can also solve it with `take` function. You already found how many elements you need to take (by the way, nice job putting it into `where` construction, very readable 👍🏼 ).